src/js/UrbanAirship.js

```diff
@@ -324,6 +324,29 @@ class UrbanAirship {
     });
   }
 
+  /**
+   * Creates an editor to modify the named user attributes.
+   *
+   * @return {Promise.<AttributeEditor, Error>}  A promise that returns an attribute editor instance if resolved, or an Error if the
```

Comment:
The SDKs allow attributes to be set before setting a named user id. The attributes are uploaded once a named user id has been set. Why did you decide to force the named user id to be set before setting attributes?

Reply:
I was following the ticket asking to return an error in the callback if the named user is not set.

Reply:
Well, that's weird. Why would we do that in frameworks if the SDKs don't do it? I feel like we should get @rlepinski to clarify.